### PR TITLE
EG-1296 Implement Sequence UI component and render view

### PR DESCRIPTION
**Pull request by woollett-r**
Note, I've used the SubmittedTextAnswerFormProvider for the time being; this may change. I have also used the identifier name[index] for the check box identifiers. I think this will work with the "list" type binding, but I'll find out more later.

### CODE REVIEW

File: app/forms/FormsHelper.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ object FormsHelper {
   def populatedForm(inputStanza: DataInput, path: String, answer: Option[String]): Form[_] =
     inputStanza match {
       case _: DateInput => populateSubmittedDateAnswerForm(answer)
-      case _: Input | _: Question => populateSubmittedTextAnswerForm(path -> nonEmptyText, path, answer)
+      case _: Input | _: Question | _: Sequence => populateSubmittedTextAnswerForm(path -> nonEmptyText, path, answer) // TODO - Temporary solution whilst rendering sequence
```
Comment (cmrhmrc) [inline app/forms/FormsHelper.scala:39]: think this will also need adding to bindFormData above

File: app/services/UIBuilder.scala [hunk 2]
```diff
@@ -196,4 +197,14 @@ class UIBuilder {
   private def fromSectionAndNoteCallout(caption: Text, nc: NoteCallout)(implicit stanzaIdToUrlMap: Map[String, String], lang: Lang): UIComponent =
     Details(caption, Seq(TextBuilder.fromPhrase(nc.text)))
 
+  private def fromSequence(sequence: Sequence, components: Seq[UIComponent])
+                          (implicit stanzaIdToUrlMap: Map[String, String], lang: Lang): UIComponent = {
+    val (errMsgs, uiElements) = partitionComponents(components, Seq.empty, Seq.empty)
+
+    val (text, hint) = TextBuilder.singleTextWithOptionalHint(sequence.text)
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:204]: There is a bug in Question (in the backlog) to replace use of singleTextWithOptionalHint with something like fromPhrase which also extracts the hint. Inputs dont have this bug, it prevents what becomes the H1 from including label references (this is one of Abby's discoveries)

File: app/services/UIBuilder.scala [hunk 3]
```diff
@@ -196,4 +197,14 @@ class UIBuilder {
   private def fromSectionAndNoteCallout(caption: Text, nc: NoteCallout)(implicit stanzaIdToUrlMap: Map[String, String], lang: Lang): UIComponent =
     Details(caption, Seq(TextBuilder.fromPhrase(nc.text)))
 
+  private def fromSequence(sequence: Sequence, components: Seq[UIComponent])
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:200]: Looking at GDS, it supports hints on each checkbox item, so we'll need to support the [hint:....] label at the end of each checkbox item text as the main text

File: app/forms/FormsHelper.scala (hunk 1 above)
Comment (woollett-r) [reply app/forms/FormsHelper.scala:39]: This will be done as part of the submission task.

File: app/services/UIBuilder.scala (hunk 3 above)
Comment (woollett-r) [reply app/services/UIBuilder.scala:200]: I have raised separate task for this (EG-1309)

File: app/services/UIBuilder.scala (hunk 2 above)
Comment (woollett-r) [reply app/services/UIBuilder.scala:204]: I've updated task EG-1307 which records this defect such that sequence should also be fixed.

**Pull request merged**